backend/src/apiserver/server/util.go

```diff
@@ -76,18 +80,23 @@ func DecompressPipelineTarball(compressedFile []byte) ([]byte, error) {
 		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the tarball file. Not a valid tarball file.")
 	}
 	tarReader := tar.NewReader(gzipReader)
-	header, err := tarReader.Next()
-	if err != nil || header == nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the tarball file. Not a valid tarball file.")
-	}
-	if !isYamlFile(header.Name) {
-		return nil, util.NewInvalidInputError("Error extracting pipeline from the tarball file. Expecting a YAML file inside the tarball. Got: %v", header.Name)
-	}
-	decompressedFile, err := ioutil.ReadAll(tarReader)
-	if err != nil {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error reading pipeline YAML from the tarball file.")
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil || header == nil {
+			return nil, util.NewInvalidInputErrorWithDetails(err, "Error extracting pipeline from the tarball file. Not a valid tarball file.")
+		}
+		if isPipelineYamlFile(header.Name) {
+			decompressedFile, err := ioutil.ReadAll(tarReader)
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(err, "Error reading pipeline YAML from the tarball file.")
+			}
+			return decompressedFile, err
+		}
 	}
-	return decompressedFile, err
+	return nil, util.NewInvalidInputError("Error extracting pipeline from the .tar.gz file. Could not find any *pipeline.yaml files inside the archive.")
```

Comment:
Just to be safe. Instead of throwing here, it's better to fallback to old behavior by using the first file in the package. In this way, we can guarantee no regression.

Reply:
I'll do that, thanks for the idea.

I wonder, should we check the number of files in the package and base the behavior on whether this is a single-file or multi-file package?

Reply:
Better to be some as old behavior: just use the first file.

Reply:
Done.